compiler/rustc_const_eval/src/interpret/cast.rs

```diff
@@ -108,6 +108,19 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                     _ => span_bug!(self.cur_span(), "closure fn pointer on {:?}", src.layout.ty),
                 }
             }
+
+            DynStar => {
+                if let ty::Dynamic(data, _, ty::DynStar) = cast_ty.kind() {
+                    // Initial cast from sized to dyn trait
+                    let vtable = self.get_vtable_ptr(src.layout.ty, data.principal())?;
+                    let vtable = Scalar::from_maybe_pointer(vtable, &*self.tcx);
+                    let ptr = self.read_immediate(src)?.to_scalar();
```

Comment:
Is this always a pointer? I don't know which types are accepted for these casts. But the DynStar docs elsewhere say this can be a pointer or inline data, and the codegen_ssa equivalent of this calls it `data`.

Reply:
I renamed this to data to match the other equivalent.